doxygen/src/pack_publish.txt

```diff
@@ -1,75 +1,225 @@
+  \elem{url} location, an (unversioned) \elem{vendor}.\elem{name}.pdsc file and a (versioned)
+  \elem{vendor}.\elem{name}.\elem{version}.pack must be available.
+- The \elem{url} is the page where the pack is downloaded. This means, if the URL or the PDSC/pack files become unavailable, users are unable to download the pack.
+- When a new PDSC/pack file is available, it is important to update the version number, otherwise the \ref cp_KeilComPack "Pack Index Service" will not recognize that the pack has changed.
+- Previous versions of a (versioned) \elem{vendor}.\elem{name}.\elem{version}.pack should remain available.
+
+\subsection cp_GitHubHosting GitHub-hosted packs
+
+If you develop your packs publicly on GitHub, you can use the infrastructure to host your packs. The recommended way to do
+this is as follows (note that in this example, \elem{orgname} is equal to \elem{vendor} in the other examples. We use
+\elem{orgname} here as this is the nomenclature on GitHub):
+- Create a repository for the PIDX file, for example \c github.com/orgname/indexrepo. The PIDX file should be named \elem{orgname}.pidx.
+- Create seperate repositories for each pack, for example:
+  - \c github.com/orgname/name1
+  - \c github.com/orgname/name2
+- At the top-level of each repository's \c main branch, you need have a \elem{orgname}.\elem{nameX}.pdsc file.
+- Once you publish a pack, attach the \elem{orgname}.\elem{nameX}.\elem{version}.pack file to the release assets - this will be
+  used as the download location of the pack.
+
+This diagram shows an exemplary set up on GitHub with pack versions 1.0.0 and 1.3.0:
+
+\image html GHhosting.png "Hosting packs on GitHub"
+
+\subsubsection gh_pidx PIDX file on GitHub
+
+The vendor (orgname) PIDX file would look like the following:
+
+\code{.xml}
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<index schemaVersion="1.0.0" xs:noNamespaceSchemaLocation="PackIndex.xsd" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance">
+  <vendor>orgname</vendor>
+  <url>https://github.com/orgname/indexrepo/raw/main/</url>
+  <timestamp>2024-08-22 10:08:36</timestamp>
+  <pindex>
+    <pdsc url="https://github.com/orgname/name1/raw/main/" vendor=”orgname" name=”name1" version="1.0.0"/>
+    <pdsc url="https://github.com/orgname/name2/raw/main/" vendor=”orgname" name=”name2" version="1.0.0"/>
+  </pindex>
+</index>
+\endcode
+
+\note When updating one of the packs, you only need to change the version number. All other entries remain the same.
+
+\subsubsection gh_pdsc PDSC file on GitHub
+
+A PDSC file for one of the repositories would look like this:
+
+\code{.xml}
+<?xml version="1.0" encoding="utf-8"?>
+
+<package schemaVersion="1.7.7" xmlns:xs="http://www.w3.org/2001/XMLSchema-instance" xs:noNamespaceSchemaLocation="PACK.xsd">
+  <vendor>orgname</vendor>
+  <name>name1</name>
+  <license>LICENSE</license>
+  <description>Whatever Pack</description>
+  <url>https://github.com/orgname/name1/releases/download/v1.0.0/</url>
```

Comment:
The <url> element in the pdsc file must remain constant and must not have a version!
We had agreed that this should point to the raw file in the main branch except that the file will be composed from `<vendor>.<name>.pdsc.`
<url>https://github.com/orgname/name1/raw/main/</url>